package resolver;
option cc_generic_services = true;
option java_generic_services = true;
option py_generic_services = true;

message ResolveRequest {
  required string address = 1;
  required string name = 2;
  required int32 age = 3;
  required int32 number = 4;
}

message ResolveResponse {
  optional bool resolved = 1 [default=false];
  repeated fixed32 ip = 2;
  repeated int32 port = 3;
}

service ResolverService {
  rpc Resolve (ResolveRequest) returns (ResolveResponse);
}

